riot/riot.py

```diff
@@ -374,16 +409,35 @@ def run(
                 instance=inst, venv_name=venv_path, pkgstr=pkg_str
             )
 
+            # Generate the environment for the instance.
+            if pass_env:
+                env = os.environ.copy()
+            else:
+                env = {}
+
+            # Add in the instance env vars.
+            env.update(dict(inst.env))
+
+            if python_path:
+                if "PYTHONPATH" in os.environ:
```

Comment:
Do we actually want this? I feel like it could lead to some baffling behaviour as a user because packages I don't specify in my riot virtualenv might be locatable through a sneaky PYTHONPATH environment variable being set in my shell.

Did you have a use-case in mind here? I think there might be cases for it but I'd prefer it be opt-in.

Reply:
I was only thinking for compatibility, but makes sense to fully isolate and ignore any globally set env variables.